GitLab forge: get_repo_labels() doesn't paginate, causing 409 on label create when project label count exceeds page size

# Summary
On GitLab projects with more than ~20 labels (including labels inherited from parent groups), releasaurus release-pr consistently fails on the label replace step with:

```
[INFO] Replacing labels on PR #N with: ["releasaurus::pending"]
[ERROR] Failed to update labels on PR: GitLab API error: gitlab server error (409 Conflict): Label already exists
```

The release MR itself is created/updated successfully — the failure is only on the final label-replace step, but it red-bars the pipeline.

### Steps to Reproduce
1. Use a GitLab project that already has >20 labels, or run releasaurus a couple of times so both `releasaurus:pending` (legacy) and `releasaurus::pending` (scoped) end up in the project's labels.
2. Run `releasaurus release-pr --forge gitlab --repo <url>`
3. Job fails with 409 Conflict on label create.

Observed on v0.18.0 and v0.13.0.

### Root cause

In `crates/core/src/forge/gitlab.rs`:

```rust
async fn get_repo_labels(&self) -> Result<Vec<LabelInfo>> {
    let endpoint = Labels::builder().project(&self.project_id).build()?;
    let labels: Vec<LabelInfo> = endpoint.query_async(&self.gl).await?;
    Ok(labels)
  }
```

`Labels::builder()` returns a single page (GitLab default: 20 entries). replace_pr_labels then does:

```rust
if let Some(label) = all_labels.iter().find(|l| l.name == name) {
    labels.push(label.name.clone());
  } else {
    let label = self.create_label(name).await?;   // 409 here
```

If the target label exists but isn't on page 1, the existence check returns None, `create_label` is called, and GitLab rejects with 409.

### Suggested fix

Use the gitlab crate's pagination helper to iterate all pages, e.g.:

```rust
use gitlab::api::Pagination;
    let labels: Vec<LabelInfo> = gitlab::api::paged(endpoint, Pagination::All)
    .query_async(&self.gl).await?;
```

Optionally also tolerate 409 from create_label as a soft-success (re-fetch and use the existing label) for defense in depth.

--- a/crates/core/src/forge/gitlab.rs
+++ b/crates/core/src/forge/gitlab.rs
@@ -134,8 +134,9 @@ impl Gitlab {
     async fn get_repo_labels(&self) -> Result<Vec<LabelInfo>> {
         let endpoint = Labels::builder().project(&self.project_id).build()?;
 
-        let labels: Vec<LabelInfo> =
-            paged(endpoint, Pagination::All).query_async(&self.gl).await?;
+        let labels: Vec<LabelInfo> = paged(endpoint, Pagination::All)
+            .query_async(&self.gl)
+            .await?;
 
         Ok(labels)
     }

--- a/crates/core/src/forge/gitlab.rs
+++ b/crates/core/src/forge/gitlab.rs
@@ -134,7 +134,8 @@ impl Gitlab {
     async fn get_repo_labels(&self) -> Result<Vec<LabelInfo>> {
         let endpoint = Labels::builder().project(&self.project_id).build()?;
 
-        let labels: Vec<LabelInfo> = endpoint.query_async(&self.gl).await?;
+        let labels: Vec<LabelInfo> =
+            paged(endpoint, Pagination::All).query_async(&self.gl).await?;
 
         Ok(labels)
     }

